ie_serving/main.py

```diff
@@ -64,22 +65,46 @@ def parse_config(args):
     check_config_structure(configs=configs)
     models = {}
     for config in configs['model_config_list']:
-        if 'batch_size' in config['config'].keys():
-            batch_size = config['config']['batch_size']
-        else:
-            batch_size = None
-        model = ModelBuilder.build(model_name=config['config']['name'],
-                                   model_directory=config
-                                   ['config']['base_path'],
-                                   batch_size=batch_size)
-        models[config['config']['name']] = model
+        try:
+            batch_size = config['config'].get('batch_size', None)
+            model_ver_policy = config['config'].get(
+                'model_version_policy', None)
+            model = ModelBuilder.build(model_name=config['config']['name'],
+                                       model_directory=config['config'][
+                                           'base_path'],
+                                       batch_size=batch_size,
+                                       model_version_policy=model_ver_policy)
+            models[config['config']['name']] = model
+        except ValidationError as e_val:
+            logger.warning("Problem with model_version_policy in {} model. "
+                           "Exception: {}".format(config['config']['name'],
+                                                  e_val))
+        except Exception as e:
+            logger.warning("Unexpected error occurred in {} model. "
+                           "Exception: {}".format(config['config']['name'],
+                                                  e))
     start_server(models=models, max_workers=1, port=args.port)
 
 
 def parse_one_model(args):
-    model = ModelBuilder.build(model_name=args.model_name,
-                               model_directory=args.model_path,
-                               batch_size=args.batch_size)
+    try:
+        model_version_policy = json.loads(args.model_version_policy)
+        model = ModelBuilder.build(model_name=args.model_name,
+                                   model_directory=args.model_path,
+                                   batch_size=args.batch_size,
+                                   model_version_policy=model_version_policy)
+    except ValidationError as e_val:
+        logger.error("Problem with model_version_policy. "
+                     "Exception: {}".format(e_val))
+        sys.exit()
+    except json.decoder.JSONDecodeError as e_json:
+        logger.error("model_version_policy must be in json format. "
+                     "Exception: {}".format(e_json))
+        sys.exit()
+    except Exception as e:

```

Comment:
earlier, there was no try catch here, what change that now, we expect - "unexpected error"?